server/chat-plugins/chatlog.ts

```diff
@@ -518,49 +528,71 @@ const LogSearcher = new class {
 		}
 		return {results, total};
 	}
-
-	async ripgrepSearch(roomid: RoomID, search: string, limit?: number | null) {
-		let output;
-		if (!limit || limit > MAX_RESULTS) limit = MAX_RESULTS;
+	async ripgrepSearchMonth(roomid: RoomID, search: string, limit: number, date: string) {
+		let results;
 		try {
-			const options = [
-				'-e', `[^a-zA-Z0-9]${search.split('').join('[^a-zA-Z0-9]*')}([^a-zA-Z0-9]|\\z)`,
-				`${__dirname}/../../logs/chat/${roomid}`,
+			const {stdout} = await execFile('rg', [
+				'-e', this.constructRegex(search),
+				`logs/chat/${roomid}${toID(date) === 'all' ? '' : `/${date}`}`,
```

Comment:
This will break unless `date` is a month. If `date` is a month, it should be named `month`.